### PR TITLE
Navigation Editor: Introduce useMenuEntityProp hook

### LINKED ISSUE #30442: Naviagion edtior - encapsulate menu constants inside use-menu-entity hook
## What problem does this address?

Implement enhancement proposed in https://github.com/WordPress/gutenberg/pull/30340#discussion_r603858811

### PR DESCRIPTION
**Pull request by Mamaduka**
## Description
Adds `useMenuEntityProp` as discussed in https://github.com/WordPress/gutenberg/pull/30340#discussion_r604536574. This is a `useEntitytProp` wrapper hook that encapsulates menu constants and provides an easy way to access/edit individual menu properties.

P.S. Maybe we should also introduce the `useSelectedMenuId` hook to get menuId from context. Most of the time `useSelectedMenuData` hook is used for this.

Fixes #30442.

## How has this been tested?
Everything should work like before.

* When changing the menu name, changes should be visible in the heading and menu switcher.
* "Add new pages" should be saved when clicking the "Save" button.

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by Mamaduka** [comment]
I think this also fixes #30425.

**Comment by talldan** [comment]
> Maybe we should also introduce the useSelectedMenuId hook to get menuId from context.

There's also https://github.com/WordPress/gutenberg/issues/22623, which might influence how the selected menu id works. An option for solving that is to move the selected menu id to the store, and then use the persistence plugin (https://github.com/WordPress/gutenberg/tree/trunk/packages/data/src/plugins/persistence) to sync the value to local storage.

**Comment by Mamaduka** [comment]
Hi, @talldan

Updated PR to reflect changes from #31320.

**Review by getdave — COMMENTED** [review]
I tested this manually in the UI and didn't spot any regressions.

That said I'm not clear on

> "Add new pages" should be saved when clicking the "Save" button.

From the code it looks like previously when you toggled `Add new pages` in the Inspector sidebar the menu was automatically saved. Am I right in thinking this PR alters that behaviour so you now manually need to click `Save`? 

If so it feels like that might want to be a separate PR along with a rationale for why this is done. I may have the wrong end of this stick however 😄 

With a little more info I would be happy to approve this one.

**Comment by Mamaduka** [comment]
> From the code it looks like previously when you toggled Add new pages in the Inspector sidebar the menu was automatically saved. Am I right in thinking this PR alters that behaviour so you now manually need to click Save?

Sorry about that. I accidentally fixed #30425 when refactoring to use the new hook. The issue also has PR (#30446) that takes a different approach by moving state up the level and using the `useMenuEntity` hook.

Happy to remove the "Add new pages" part and create separate PR for the mentioned issue.

**Review by getdave — APPROVED** [review]
No worries. Thanks for clarifying.

> Happy to remove the "Add new pages" part and create separate PR for the mentioned issue.

Although it seems a bit tedious it might be easier to review in isolation 👍 

I'll approve this PR because I know that the saving of the toggle only happened when I clicked the `Save` button. I'll leave it to you as to whether you want to spin up a separate PR instead.

**Comment by Mamaduka** [comment]
Hi, @getdave

I reverted changes for "Add new pages" and will create a separated PR in a bit.
I am going to merge this one once all checks are green.

Thanks again for the review.

**Comment by getdave** [comment]
> I reverted changes for "Add new pages" and will create a separated PR in a bit.

Sure thing. Let me know and I'll happily review.

**Comment by Mamaduka** [comment]
Thanks, @getdave

Here's new PR #31643.

**Pull request merged**
